src/main/groovy/de/dkfz/roddy/execution/jobs/SubmissionCommand.groovy

```diff
@@ -34,24 +49,20 @@ abstract class SubmissionCommand extends Command {
     }
 
     /**
+     * Should the local environment be passed?
+     *
      * JobManager and SubmissionCommand together determine, whether the environment should be passed.
      *
-     * * If the Command value is set to true or false, the JobManager value is overruled.
+     * * The Command has precedence of the JobManager value.
      * * If the Command value is not set, the JobManager value is the fallback.
-     * * If neither of JobManager and Command is defined, return false.
+     * * If neither of JobManager and Command is defined, return only copy the requested variables to remote.
      *
      * @return
      */
-    Optional<Boolean> getPassLocalEnvironment() {
-        if (passEnvironment.present) {
-            return passEnvironment
-        } else {
-            if (parentJobManager.passEnvironment.present) {
-                return parentJobManager.passEnvironment
-            } else {
-                return Optional.empty()
-            }
-        }
+    PassEnvironmentVariables getPassLocalEnvironment() {
+        passEnvironment.
```

Comment:
Interesting construct :D personally I'd prefer the if/else stuff over this. But ok, leave it.

Reply:
Functional style using the Optional DSL.